pkg/controllers/management/node/controller.go

```diff
@@ -95,15 +107,7 @@ func (m *Lifecycle) setWaiting(node *v3.Node) {
 
 func (m *Lifecycle) Create(obj *v3.Node) (*v3.Node, error) {
 	if isCustom(obj) {
-		m.setupCustom(obj)
-		newObj, err := v3.NodeConditionInitialized.Once(obj, func() (runtime.Object, error) {
-			if err := validateCustomHost(obj); err != nil {
-				return obj, err
-			}
-			m.setWaiting(obj)
-			return obj, nil
-		})
-		return newObj.(*v3.Node), err
+		return m.provisionCustom(obj)
```

Comment:
Although we know that the update is being called right after create, it is better to set NodeConditionInitialized in Create only. And do it similarly to how it was done before - only when validateCustomHost/setWaiting. Unless there was a specific reason for moving that part under update?